garrysmod/gamemodes/terrortown/entities/entities/ttt_logic_role.lua

```diff
@@ -24,7 +24,7 @@ end
 function ENT:AcceptInput(name, activator)
    if name == "TestActivator" then
 
-      if IsValid(activator) and activator:IsPlayer() and (self.Role == ROLE_ANY or activator:IsRole(self.Role)) then
+      if IsValid(activator) and activator:IsPlayer() and (self.Role == ROLE_ANY or activator:IsActiveRole(self.Role)) then

```

Comment:
Just using IsActiveRole was my first thought as well, but it's not ideal. It will return false even if the `ttt_logic_role` is checking for ROLE_INNOCENT, if it's during the prep or post round. Similarly a check for traitors will be false if it's post round, so a traitor in a traitor-room would be stuck in there for the whole post-round phase.

Perhaps something like this

    local activator_role = (GetRoundState() == ROUND_PREP) and ROLE_INNOCENT or activator:GetRole()

Which you could then compare to `self.Role`. Of course you'd also need to do the IsPlayer check before that.